### ISSUE TITLE
Regex containing a bracket fails

### ISSUE THREAD

**Issue by szepeviktor** [douglascrockford/JSLint#205]
```js
var pear = "pear";
var output;

// Fails
output = pear.replace(/{([^{}]*)}/g, "pear");
// OK
output = pear.replace(/([^{}]*)/g, "pear");
```

Please advise.

**Comment by szepeviktor** [comment]
Code is from your video: http://youtu.be/RO1Wnu-xKoY?t=58m41s

**Comment by douglascrockford** [comment]
output = pear.replace(/\{([^{}]*)\}/g, "pear");

**Comment by szepeviktor** [comment]
Yes. This one fails.
INPUT
```js
var pear = "pear";
var output;

// Fails
output = pear.replace(/{([^{}]*)}/g, "pear");
```
OUTPUT
```
JSLint was unable to finish.
4.23
Expected '/' and instead saw '{'.
output = pear.replace(/{([^{}]*)}/g, "pear");
4.24
Expected a regexp factor and instead saw '{'.
output = pear.replace(/{([^{}]*)}/g, "pear");
```

**Comment by szepeviktor** [comment]
This is most *JSLint* version of `String.supplant()`
https://github.com/szepeviktor/wordpress-sitebuild/blob/master/js/String.supplant.js

**Comment by douglascrockford** [comment]
Thanks. Please try it now.

**Comment by szepeviktor** [comment]
Thank you.
I've seen your commit https://github.com/douglascrockford/JSLint/commit/3575229b0894e36b056a7bbd845c93f88e6aa695
But I am sorry to report that pasting [the above code](https://github.com/douglascrockford/JSLint/issues/205#issuecomment-278160827) to jslint.com yields another error message.
```
4.23
Expected '\' before '{'.
output = pear.replace(/{([^{}]*)}/g, "pear");
4.32
Expected '\' before '}'.
output = pear.replace(/{([^{}]*)}/g, "pear");
```

**Comment by szepeviktor** [comment]
I think I should escape brackets in the regexp `output = pear.replace(/\{([^{}]*)\}/g, "pear");`

You know: I've just followed your video :)
Thanks!

**Comment by douglascrockford** [comment]
JSLint's advice is good.
